src/kontrol/prove.py

```diff
@@ -1147,10 +1147,55 @@ def extend_storage(map: KInner, slot: int, value: KInner) -> KInner:
                             '_Map_', [map_item(intToken(field.slot), contract_account_variable), storage_map]
                         )
                     else:
-                        # TODO: Support shared slots for contract variables
-                        raise (
-                            ValueError(
-                                'Unsupported: CSE for contracts with contract or interface fields in shared slots.'
+                        slot_var_before = KVariable(f'{field_name}_SLOT_BEFORE', sort=KSort('Bytes'))
+                        slot_var_after = KVariable(f'{field_name}_SLOT_AFTER', sort=KSort('Bytes'))
+                        masked_contract_account_var = KApply(
+                            'asWord',
+                            [
+                                KApply(
+                                    '_+Bytes__BYTES-HOOKED_Bytes_Bytes_Bytes',
+                                    [
+                                        KApply(
+                                            '_+Bytes__BYTES-HOOKED_Bytes_Bytes_Bytes',
+                                            [
+                                                slot_var_after,
+                                                KEVM.buf(intToken(20), contract_account_variable),
+                                            ],
+                                        ),
+                                        slot_var_before,
```

Comment:
Please have `before` on the LHS and `after` on the RHS since the indexing of `+Bytes` is left-to-right, and also have the `+Bytes` be right-associative so that no further simplifications are needed to transform this into the canonical form.

Reply:
The reason I have `before` on the right and `after` on the left is because I was trying to refer to the order in which the storage fields are be declared in the contract, so `before` would be those fields in the same slot declared before the contract and `after` those declared after. Does it not do that or do you not like that convention?

Reply:
I changed it so the `+Bytes` term is right-grouped to start.

Reply:
Since `+Bytes` indexing is always left-to-right, for me 'before' in `Bytes` means 'to the left' (not sure about others), that's all, and since we will be debugging code like this likely often, I'm thinking that it's better to not break that intuition.

Reply:
Ok, thanks. I've changed the ordering.